Add NCCL Alltoall to PT NCCL process group

srinivas212 opened the pull request:
Differential Revision: D22917967

mrshenli reviewed (approved):
Test errors are irrelevant. Thanks for adding this!

```
Aug 04 06:24:51 ERROR test/onnx/test_pytorch_onnx_onnxruntime.py - NameError: name 'enableScr...
Aug 04 06:24:51 ERROR test/onnx/test_pytorch_onnx_onnxruntime.py - NameError: name 'enableScr...
Aug 04 06:24:51 ERROR test/onnx/test_pytorch_onnx_onnxruntime.py - NameError: name 'enableScr...
```

Commit:
Add NCCL Alltoall to PT NCCL process group (#42514)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/42514

Add Alltoall and Alltoallv to PT NCCL process group using NCCL Send/Recv.

Reviewed By: mrshenli

Differential Revision: D22917967

fbshipit-source-id: e3345ddbb5217dc5fabcd5c02db4cf67becff4b5
